subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/MinimalJavaCompileOptions.java

```diff
@@ -49,7 +49,7 @@
     public MinimalJavaCompileOptions(final CompileOptions compileOptions) {
         FileCollection sourcepath = compileOptions.getSourcepath();
         this.sourcepath = sourcepath == null ? null : ImmutableList.copyOf(sourcepath.getFiles());
-        this.compilerArgs = Lists.newArrayList(compileOptions.getCompilerArgs());
+        this.compilerArgs = Lists.newArrayList(compileOptions.getAllCompilerArgs());
         this.encoding = compileOptions.getEncoding();
         this.bootClasspath = DeprecationLogger.whileDisabled(new Factory<String>() {
```

Comment:
Unrelated, but: why is this using the deprecated `bootClasspath` instead of the new `bootstrapClasspath`?

Reply:
I guess we use `bootClasspath` since it returns the right `String`.
`bootstrapClasspath` is a `FileCollection`, which cannot be serialized.